src/Collision.cc

```diff
@@ -256,23 +262,59 @@ void Collision::CalculateInertial(
   gz::math::Inertiald &_inertial,
   const ParserConfig &_config)
 {
-  // Check if density was not set during load & send a warning
-  // about the default value being used
-  if (!this->dataPtr->densitySetAtLoad)
+  this->CalculateInertial(
+    _errors, _inertial, _config, std::nullopt, ElementPtr());
+}
+
+/////////////////////////////////////////////////
+void Collision::CalculateInertial(
+  sdf::Errors &_errors,
+  gz::math::Inertiald &_inertial,
+  const ParserConfig &_config,
+  const std::optional<double> &_density,
+  sdf::ElementPtr _autoInertiaParams)
+{
+  // Order of precedence for density:
+  double density;
+  // 1. Density explicitly set in this collision, either from the
+  // `//collision/density` element or from Collision::SetDensity.
+  if (this->dataPtr->density)
+  {
+    density = *this->dataPtr->density;
+  }
+  // 2. Density passed into this function, which likely comes from the
+  // `//link/inertial/density` element or from Link::SetDensity.
+  else if (_density)
   {
+    density = *_density;
+  }
+  // 3. DensityDefault value.
+  else
+  {
+    // If density was not explicitly set, send a warning
+    // about the default value being used
+    density = DensityDefault();
     Error densityMissingErr(
       ErrorCode::ELEMENT_MISSING,
       "Collision is missing a <density> child element. "
-      "Using a default density value of 1000.0 kg/m^3. "
+      "Using a default density value of " +
+      std::to_string(DensityDefault()) + " kg/m^3. "
     );
     enforceConfigurablePolicyCondition(
       _config.WarningsPolicy(), densityMissingErr, _errors
     );
   }
 
+  // If this Collision's auto inertia params have not been set, then use the
+  // params passed into this function.
+  sdf::ElementPtr autoInertiaParams = this->dataPtr->autoInertiaParams;
```

Comment:
We should document on sdf_tutorials that this works by completely replacing the parameters in the link, i.e, the values inside `auto_inertia_params` of the collision do not "merge" with the ones lin the parent link.

Reply:
sure, let's document how it's implemented. I suppose we could merge the parameters in some way but we'd need to define the merge strategy first. the current approach is simplest

Reply:
Yes, I think the current approach is simplest and since we don't have anything currently using these, let's hold off on the merge idea. I just wanted to make sure we document the behavior since it's easy for users to think the values are being merged.